docs/source/howto/exploring.rst

```diff
@@ -0,0 +1,137 @@
+
+If there is no neighboring output with the given link label, a :class:`~aiida.common.exceptions.NotExistentAttributeError` or :class:`~aiida.common.exceptions.NotExistentKeyError` will be raised, respectively.
+
+.. note::
+
+    The ``inputs`` and ``outputs`` properties are only defined for :class:`~aiida.orm.nodes.process.process.ProcessNode`'s.
+    This means that you cannot *chain* these calls, because an input or output of a process node is guaranteed to be a :class:`~aiida.orm.nodes.data.Data` node, which does not have inputs or outputs.
+
+
+.. _how-to:exploring:creator-caller-called:
+
+Creator, caller and called
+==========================
+
+Similar to the ``inputs`` and ``outputs`` properties of process nodes, there are some more properties that make exploring the provenance graph easier:
+
+    * :meth:`~aiida.orm.nodes.process.process.ProcessNode.called`: defined for :class:`~aiida.orm.nodes.process.process.ProcessNode`'s and returns the list of process nodes called by this node.
+      If this process node did not call any other processes, this property returns an empty list.
+    * :meth:`~aiida.orm.nodes.process.process.ProcessNode.caller`: defined for :class:`~aiida.orm.nodes.process.process.ProcessNode`'s and returns the process node that called this node.
+      If this ndoe was not called by a process, this property returns ``None``.
+    * :meth:`~aiida.orm.nodes.data.Data.creator`: defined for :class:`~aiida.orm.nodes.data.Data` nodes and returns the process node that created it.
+      If the node was not created by a process, this property returns ``None``.
+
+
+.. _how-to:exploring:calcjob-results:
+
+Calculation job results
+=======================
+
+:class:`~aiida.orm.nodes.process.calculation.calcjob.CalcJobNode`'s provide the :meth:`~aiida.orm.nodes.process.calculation.calcjob.CalcJobNode.res` property, that can give easy access to the results of the calculation job.
+The requirement is that the :class:`~aiida.engine.processes.calcjobs.calcjob.CalcJob` class that produced the node, defines a *default output node* in its spec.
+This node should be a :class:`~aiida.orm.nodes.data.dict.Dict` output that will always be created.
+An example is the :class:`~aiida.calculations.templatereplacer.TemplatereplacerCalculation` plugin, that has the ``output_parameters`` output that is specified as its default output node.
+
+The :meth:`~aiida.orm.nodes.process.calculation.calcjob.CalcJobNode.res` property will give direct easy access to all the keys within this dictionary output.
+For example, the following:
+
+.. code-block:: python
+
+    list(node.res)
+
+will return a list of all the keys in the output node.
+Individual keys can then be accessed through attribute dereferencing:
+
+.. code-block:: python
+
+    node.res.some_key
+
+In an interactive shell, the available keys are also tab-completed.
+If you type ``node.res.`` followed by the tab key twice, a list of the available keys is printed.
```

Comment:
I would add that, if the default output node link label is `output_parameters`, then `node.res.some_key` is equivalent to `node.outputs.output_parameters.dict.some_key` and stress that we are looking at the keys of a different node (this, together with the fact that `verdi node outputcat/outputls` act on a different node, does not seem to be super intuitive at first for new users, in my experience)